ArduPlane/mode_acro.cpp

```diff
@@ -1,4 +1,3 @@
-#include "mode.h"
```

Comment:
This isn't right.

You're now relying on Plane to give you the Mode definitions, which isn't the right way to get them.

Similarly elsewhere.

Getting rid of the Plane.h include would be nice, 'though :-)  Rather unlikely, however, as most will be doing `plane.` something

Reply:
The issue I have is that if `mode.h` is included here, `SOARING_ENABLED` is not yet defined within `mode.h`.

If I include `config.h` in `mode.h`, `AP_AHRS_NAVEKF_AVAILABLE` is not yet defined within `config.h`.

I'll spend some more time on this - any pointers appreciated!